src/modal/alert-modal.interface.ts

```diff
@@ -48,6 +48,12 @@ export interface AlertModalData {
 	 * Callback for non-specific close events. `return false;` to prevent the modal from closing
 	 */
 	close?: Function;
```

Comment:
Perhaps changing the signature to be something like:
```suggestion
	close?: ({ trigger: "overlay" | "close" }) => boolean;
```
Would be more flexible, and perhaps a touch more explicit with regard to what we're actually handling